transport/sqs/src/test/kotlin/SqsMessageSenderFactoryTest.kt

```diff
@@ -0,0 +1,136 @@
+import org.eclipse.apoapsis.ortserver.transport.json.JsonSerializer
+import org.eclipse.apoapsis.ortserver.transport.testing.TEST_QUEUE_NAME
+
+class SqsMessageSenderFactoryTest : StringSpec({
+    val consumerName = "orchestrator"
+    val transportType = "sender"
+
+    val configManager = createSqsConfigManager(consumerName, transportType)
+    val client = createSqsClient(SqsConfig.create(configManager.subConfig(Path("$consumerName.$transportType"))))
+
+    val serializer = JsonSerializer.forType<OrchestratorMessage>()
+
+    "Messages can be sent via the sender" {
+        val header = MessageHeader("token", "traceId", 47)
+        val payload = AnalyzerWorkerResult(11)
+        val message = Message(header, payload)
+
+        client.createQueue(CreateQueueRequest { queueName = TEST_QUEUE_NAME })
+
+        val sender = MessageSenderFactory.createSender(OrchestratorEndpoint, configManager)
+        sender.send(message)
+
+        val queueResponse = client.getQueueUrl(GetQueueUrlRequest { queueName = TEST_QUEUE_NAME })
+
+        val receiveRequest = ReceiveMessageRequest {
+            queueUrl = queueResponse.queueUrl
+
+            // Opt-in to the message attributes to receive.
+            messageAttributeNames = org.eclipse.apoapsis.ortserver.transport.sqs.messageAttributeNames
+
+            // Enable "long polling" to eliminate empty responses, see
+            // https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/working-with-messages.html#setting-up-long-polling
+            waitTimeSeconds = 1
+        }
+
+        val receiveResponse = client.receiveMessage(receiveRequest)
+
+        receiveResponse.messages.shouldNotBeNull().run {
+            shouldHaveSize(1)
+
+            first().messageAttributes.shouldNotBeNull().run {
+                toMessageHeader() shouldBe header
+            }
+
+            first().body.shouldNotBeNull().run {
+                serializer.fromJson(this) shouldBe payload
+            }
+        }
+    }
+
+    "The client is closed by the AutoClosable implementation" {
+        val mockClient = mockk<SqsClient> {
+            every { close() } just runs
+        }
+
+        val sender = SqsMessageSender(
+            client = mockClient,
+            queueUrl = "",
+            to = AnalyzerEndpoint
+        )
+
+        sender.close()
+
+        verify { mockClient.close() }
+    }
+
+    "The client is closed if the sender cannot be created" {
+        val mockClient = spyk(client) {
+            coEvery { getQueueUrl(any()) } throws IOException()
+        }
+
+        mockkStatic(::createSqsClient)
+        every { createSqsClient(any()) } returns mockClient
+
+        client.createQueue(CreateQueueRequest { queueName = TEST_QUEUE_NAME })
+
+        shouldThrow<IOException> {
+            MessageSenderFactory.createSender(OrchestratorEndpoint, configManager)
+        }
+
+        verify { mockClient.close() }
+
+        unmockkStatic(::createSqsClient)
```

Comment:
This is not called if the test fails before which could potentially affect other tests. We usually call `unmockkAll()` in `afterTest` or `afterSpec` to prevent that. `try/finally` is also an option, but we only use it in helper functions.

Reply:
I went for the `afterEach` approach.